doc/userdoc/contribute/styleguide/styleguide.rst

```diff
@@ -0,0 +1,634 @@
+
+
+.. note::
+
+   reStructuredText is sensitive to indentation and newlines.
+
+   * Directives, headings, labels, and tables should be separated from other text by a newline, excluding directive options.
+   * Directive options must be indented to the same level as the directive content.
+   * Text in multiline content should stay aligned with the first line.
+   * If the output format seems incorrect, double check the spaces and newlines
+     of the text.
+
+
+Code and code snippets
+~~~~~~~~~~~~~~~~~~~~~~
+
+Code blocks are written by using the code-block directive.
+
+Example syntax
+
+   .. code-block:: none
+
+    .. code-block:: cpp
+
+        int main()
+        {
+          std::cout << "Hello, World!";
+          return 0;
+        }
+
+
+Code is rendered as
+
+    .. code-block:: cpp
+
+        int main()
+        {
+          std::cout << "Hello, World!";
+          return 0;
+        }
+
+
+For Python examples that show input and output, use the following syntax::
+
+   >>> input begins with 3, right-angled brackets
+   Output is directly below input without any brackets.
+   A blank line must end the example.
+
+For in-text code use the role :code: or double back ticks::
+
+   ``cout << "Hello, World!`` or
+   :code:`cout << "Hello, World!"`
+
+.. _math_style:
+
+Math equations
+~~~~~~~~~~~~~~
+
+The input language for mathematics is LaTeX markup. See `Mathematics into Type
+<http://www.ams.org/arc/styleguide/mit-2.pdf>`_ for a guide to styling LaTeX math.
+
+
+For equations that take a whole line (or more), use the math directive::
+
+    .. math::
+
+        f(x) = \int_{-\infty}^{\infty} \hat{f}(\xi) e^{2 \pi i x \xi} \, d\xi.
+
+Output rendered as
+
+    .. math::
+
+        f(x) = \int_{-\infty}^{\infty} \hat{f}(\xi) e^{2 \pi i x \xi} \, d\xi.
+
+If the equation runs over several lines you can use double backslashes ``\\`` as a separator at the end of each line.
+You can also align lines in an equation by using the ``&`` where you want an equation aligned::
+
+
+    .. math::
+
+         (a + b)^2  &=  (a + b)(a + b) \\
+                    &=  a^2 + 2ab + b^2
+
+For in-text math, use the math role::
+
+   Now we can see :math:`x=1` for this example.
+
+These will be rendered as
+
+   Now we can see :math:`x=1` for this example.
+
+.. _sec_admonition:
+
+Admonitions
+~~~~~~~~~~~
+
+Admonitions are directives that render as highlighted blocks to draw the reader's attention to a particular point.
+
+Use them sparingly.
+
+
+Use the admonition
+
+* "See also" to reference internal or external links (only in cases where the reference should stand out),
+
+* "Note" to add additional information that the reader needs to be aware of,
+
+* "Warning" to indicate that something might go wrong without the provided information, and
+
+* "Danger" if the situation may cause severe, possibly irreversible, problems.
+
+
+If you want a custom admonition, use::
+
+   .. admonition:: custom label
+
+         Here is some text
```

Comment:
For some reason, the `is` here is rendered boldface. Do you understand why and can it be avoided?

Reply:
This is because of syntax highlighting when using  the `::`, I changed it to a code-block with no highlighting